### PR TITLE
Make sure all parameter names match NumPy and CPython

### LINKED ISSUE #9053: Numpy Gamma distribution not working in Numba 0.57+
## Reporting a bug

- [ x] I have tried using the latest released version of Numba (most recent is
 visible in the change log (https://github.com/numba/numba/blob/main/CHANGE_LOG).
- [ x] I have included a self contained code sample to reproduce the problem.
  i.e. it's possible to run as 'python bug.py'.

Vectorization of Numpy's gamma distribution no longer works. The below code works in Numba 0.56.4, but not in 0.57.0 or 0.57.1. 

```
from numba import float64, vectorize
from numpy.random import gamma
import numpy as np

@vectorize([float64(float64, float64)])
def vec_ig(shape, scale):
    ig = 1. / gamma(shape=shape, scale=1. / scale)
    return ig

a = np.array([0.1, 0.1], dtype=np.float64)
b = np.array([0.1, 0.1], dtype=np.float64)
res = vec_ig(a, b)
```

The error that I receive is:

```
Compilation is falling back to object mode WITHOUT looplifting enabled because Function "vec_ig" failed type inference due to: No implementation of function Function(<built-in method gamma of numpy.random.mtrand.RandomState object at 0x7f7ae5b65840>) found for signature:
 
 >>> gamma(shape=float64, scale=float64)
 
There are 6 candidate implementations:
  - Of which 6 did not match due to:
  Overload in function 'gammavariate_impl': File: numba/cpython/randomimpl.py: Line 774.
    With argument(s): '(shape=float64, scale=float64)':
   Rejected as the implementation raised a specific error:
     TypingError: missing a required argument: 'alpha'
  raised from /home/devin/.cache/pypoetry/virtualenvs/pybuc-opbxzPBX-py3.9/lib64/python3.9/site-packages/numba/core/typing/templates.py:784
```

Expected behavior:

Return an array of size 2 with random draws from the gamma distribution.

### PR DESCRIPTION
**Pull request by apmasell**
As titled.
Closes #9053

### CODE REVIEW

File: numba/cpython/randomimpl.py [hunk 1]
```diff
@@ -625,9 +634,9 @@ def np_uniform_impl0():
 
 
 @overload(random.uniform)
-def uniform_impl1(low):
-    if isinstance(low, (types.Float, types.Integer)):
-        return lambda low: random.uniform(low, 1.0)
+def uniform_impl1(a):
+    if isinstance(a, (types.Float, types.Integer)):
+        return lambda a: random.uniform(a, 1.0)
```
Comment (guilhermeleobas) [inline numba/cpython/randomimpl.py:639]: `random.uniform` expects two arguments: `a` and `b`
https://docs.python.org/3/library/random.html#random.uniform

I guess this one is not really used/tested in Numba test suite.

**Review by guilhermeleobas — COMMENTED** [review]
Thanks, @apmasell. I've checked all the implementations and the changes match with NumPy/CPython random module. There's only one implementation that I think could be removed, as it is not used.

Also, some of the changes exceed the 80 char limit, but I guess this is fine as flake8 doesn't run in this file.

File: numba/cpython/randomimpl.py (hunk 1 above)
Comment (apmasell) [reply numba/cpython/randomimpl.py:639]: The differing number of arguments are overloaded separately. `def uniform_impl2(a, b):` is below. None of these are called by name from the test suite.

**Review by apmasell — COMMENTED** [review]
I believe all the changed lines are within the limit

File: numba/cpython/randomimpl.py (hunk 1 above)
Comment (guilhermeleobas) [reply numba/cpython/randomimpl.py:639]: Got it. But I think `random.uniform` takes two arguments: `a` and `b`. See:

```python
$ python
Python 3.10.12 | packaged by conda-forge | (main, Jun 23 2023, 22:41:52) [Clang 15.0.7 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import random
>>> random.uniform(2)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: Random.uniform() missing 1 required positional argument: 'b'
```

**Comment by guilhermeleobas** [comment]
> I believe all the changed lines are within the limit

Some lines go beyond the 80 char limit. See:

image

But I think that's fine as this file is not checked by flake8.

File: numba/cpython/randomimpl.py (hunk 1 above)
Comment (apmasell) [reply numba/cpython/randomimpl.py:639]: Okay. I can delete it, but that will go in a separate PR. It's out of scope for this change.

**Review by guilhermeleobas — APPROVED** [review]

**Comment by stuartarchibald** [comment]
Are there any tests in this PR that demonstrate that the reported issue (#9053), and associated discovered issues, are fixed?

**Comment by apmasell** [comment]
No.

**Pull request closed without merging**